src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,105 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertRequest = "INSERT INTO manufacturers (name, country) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertStatement = connection.prepareStatement(insertRequest)) {
```

Comment:
```suggestion
                PreparedStatement insertStatement = connection.prepareStatement(insertRequest, Statement.RETURN_GENERATED_KEYS))) {
```
"Use Statement.RETURN_GENERATED_KEYS only in create statement, it’s not needed in other methods."
I assume we need to generate key in this method and assign it to manufacturer.

Reply:
Right! Finally I got its purpose)